### ISSUE TITLE
IndexError when reading empty parquet file with chunksize

### ISSUE THREAD

**Issue by aloysius-lim** [dask/dask#8374]
**What happened**:

When reading an empty parquet file with `chunksize` argument, the error `"IndexError: list index out of range"` is raised. While it may seem that using `chunksize` is irrelevant, the use case here is reading files from an external source where it is not known a priori whether or not the file is empty (or really large).

**What you expected to happen**:

An empty dataframe is returned, with no errors, such as the case when `chunksize` is not used:

```python
import dask.dataframe as dd
import pandas as pd
path = "/some/path"

df = pd.DataFrame({'col1': pd.Series(dtype="int"), 'col2': pd.Series(dtype="str")})
ddf = dd.from_pandas(df, npartitions=1)
ddf.to_parquet(path, engine="pyarrow")
dd.read_parquet(path, engine="pyarrow")
```

Output:

```
Dask DataFrame Structure:
                col1    col2
npartitions=1               
               int64  object
                 ...     ...
Dask Name: read-parquet, 1 tasks
```

**Minimal Complete Verifiable Example**:

```python
import dask.dataframe as dd
import pandas as pd
path = "/some/path"

df = pd.DataFrame({'col1': pd.Series(dtype="int"), 'col2': pd.Series(dtype="str")})
ddf = dd.from_pandas(df, npartitions=1)
ddf.to_parquet(path, engine="pyarrow")
dd.read_parquet(path, engine="pyarrow", chunksize="1GB")
```

Error:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
/var/folders/4t/cc5xr5r51y944twv96s5lcsh0000gn/T/ipykernel_67656/807784452.py in <module>
----> 1 dd.read_parquet(path, engine="pyarrow", chunksize="1GB")

~/mambaforge/envs/sims/lib/python3.8/site-packages/dask/dataframe/io/parquet/core.py in read_parquet(path, columns, filters, categories, index, storage_options, engine, gather_statistics, ignore_metadata_file, metadata_task_size, split_row_groups, chunksize, aggregate_files, **kwargs)
    355 
    356     # Parse dataset statistics from metadata (if available)
--> 357     parts, divisions, index, index_in_columns = process_statistics(
    358         parts,
    359         statistics,

~/mambaforge/envs/sims/lib/python3.8/site-packages/dask/dataframe/io/parquet/core.py in process_statistics(parts, statistics, filters, index, chunksize, split_row_groups, fs, aggregation_depth)
   1156         # Aggregate parts/statistics if we are splitting by row-group
   1157         if chunksize or (split_row_groups and int(split_row_groups) > 1):
-> 1158             parts, statistics = aggregate_row_groups(
   1159                 parts, statistics, chunksize, split_row_groups, fs, aggregation_depth
   1160             )

~/mambaforge/envs/sims/lib/python3.8/site-packages/dask/dataframe/io/parquet/core.py in aggregate_row_groups(parts, stats, chunksize, split_row_groups, fs, aggregation_depth)
   1266     parts, stats, chunksize, split_row_groups, fs, aggregation_depth
   1267 ):
-> 1268     if not stats[0].get("file_path_0", None):
   1269         return parts, stats
   1270 

IndexError: list index out of range
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2021.11.1
- Python version: 3.8.12
- Operating System: macOS 11.6
- Install method (conda, pip, source): pip

**Comment by jrbourbeau** [comment]
Thanks for raising an issue @aloysius-lim -- I'm able to reproduce. It looks like `aggregate_row_groups` assumes `stats` can't be empty. Perhaps we could just pass `parts` and `stats` through if they're empty. cc @rjzamora if you have thoughts about a different approach

**Comment by rjzamora** [comment]
>It looks like aggregate_row_groups assumes stats can't be empty. Perhaps we could just pass parts and stats through if they're empty. cc @rjzamora if you have thoughts about a different approach

Thanks for the ping @jrbourbeau! You have this exactly right.  [This line](https://github.com/dask/dask/blob/f5881891505b9a2ba2da195befb11ad7b4c7bb23/dask/dataframe/io/parquet/core.py#L1268) should definitely be something like: `if not stats or not stats[0].get("file_path_0", None):`. If there are no statistics elements to iterate throug, it is completely safe to pass along the `parts` and `stats` as is.

I can certainly submit a fix with this, but perhaps i'll give it a day or so in case someone new to dask or parquet wants to give it a go.

**Comment by ian-r-rose** [comment]
This has been fixed by #8410, and is available in the `2021.12.0` release
